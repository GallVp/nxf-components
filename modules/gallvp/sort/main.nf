process SORT {
    tag "$meta.id"
    label 'process_high'
    
    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/52/52ccce28d2ab928ab862e25aae26314d69c8e38bd41ca9431c67ef05221348aa/data'
        : 'community.wave.seqera.io/library/coreutils_grep_gzip_lbzip2_pruned:838ba80435a629f8'}"

    input:
    tuple val(meta), path(input_file)

    output:
    tuple val(meta), path("*.${extension}")                   , emit: sorted
    path "versions.yml"                                       , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    extension = input_file.extension    
    if ("${input_file}" == "${prefix}.${extension}") error "Input and output names are the same, use \"task.ext.prefix\" to disambiguate!"

    """
    sort --parallel=${task.cpus} \\
        $args \\
        $input_file \\
        > ${prefix}.${extension}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sort: \$(sort --version | sed -n '/sort (GNU coreutils) / s/sort (GNU coreutils) //p')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    extension = input_file.extension
    """
    
    touch ${prefix}.${extension}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sort: \$(sort --version | sed -n '/sort (GNU coreutils) / s/sort (GNU coreutils) //p')
    END_VERSIONS
    """
}
