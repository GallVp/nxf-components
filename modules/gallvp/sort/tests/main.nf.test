nextflow_process {

    name "Test Process SORT"
    script "../main.nf"
    process "SORT"

    tag "modules"
    tag "modules_gallvp"
    tag "sort"

    test("nanopore - test_sequencing_summary") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/sequencing_summary/test.sequencing_summary.txt', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }
    test("nanopore - test_sequencing_summary stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }

}
