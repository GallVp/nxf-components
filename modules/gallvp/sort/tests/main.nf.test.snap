{
    "nanopore - test_sequencing_summary stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fc27544bdb85c9b1399a963d92304228"
                ],
                "sorted": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fc27544bdb85c9b1399a963d92304228"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-03-19T22:59:58.761013457"
    },
    "nanopore - test_sequencing_summary": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.sorted.txt:md5,56ef18b392e17b5f1dd9309f7983c867"
                    ]
                ],
                "1": [
                    "versions.yml:md5,fc27544bdb85c9b1399a963d92304228"
                ],
                "sorted": [
                    [
                        {
                            "id": "test"
                        },
                        "test.sorted.txt:md5,56ef18b392e17b5f1dd9309f7983c867"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,fc27544bdb85c9b1399a963d92304228"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2025-03-19T22:59:51.650555073"
    }
}